android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/SelectLanguageActivity.java

```diff
@@ -0,0 +1,111 @@
+
+import android.content.Context;
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.view.View;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ListAdapter;
+import android.widget.ListView;
+import android.widget.SimpleAdapter;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import com.tavultesoft.kmea.KMManager;
+import com.tavultesoft.kmea.data.Keyboard;
+import com.tavultesoft.kmea.packages.PackageProcessor;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Keyman Settings --> KeymanInstallActivity --> SelectPackageActivity --> SelectLanguageActivity
+ * Displays a list of available languages for the user to add for a given installed packageID/keyboardID.
+ */
+public final class SelectLanguageActivity extends AppCompatActivity {
+  private static final String TAG = "SelectLanguageActivity";
+  private static ArrayList<HashMap<String, String>> list = null;
+  private static Typeface titleFont = null;
+  private static final String titleKey = "title";
+  private static final String subtitleKey = "subtitle";
+  private static final String iconKey = "icon";
+  private static Context context;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
+    context = this;
+
+    setContentView(R.layout.activity_list_layout);
+    final Toolbar toolbar = findViewById(R.id.list_toolbar);
+    setSupportActionBar(toolbar);
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    getSupportActionBar().setDisplayShowHomeEnabled(true);
+    getSupportActionBar().setDisplayShowTitleEnabled(false);
+
+    final ListView listView = findViewById(R.id.listView);
+    listView.setFastScrollEnabled(true);
+
+    final Keyboard keyboard = (Keyboard)getIntent().getSerializableExtra("keyboard");
+    final String packageID = keyboard.getPackageID();
+    final String keyboardID = keyboard.getKeyboardID();
+    final String keyboardName = keyboard.getKeyboardName();
+    String title = String.format(getString(R.string.title_select_language_for_package), keyboardName);
+    final TextView textView = findViewById(R.id.bar_title);
+    textView.setText(title);
+    if (titleFont != null) {
+      textView.setTypeface(titleFont, Typeface.BOLD);
+    }
+
+    // Get the list of available languages from kmp.json
+    File resourceRoot =  new File(context.getDir("data", Context.MODE_PRIVATE).toString() + File.separator);
+    PackageProcessor kmpProcessor =  new PackageProcessor(resourceRoot);
+    List<Keyboard> availableKeyboardsList = kmpProcessor.getLanguageList(packageID, keyboardID);
+
+    list = new ArrayList<HashMap<String, String>>();
+    for (Keyboard k : availableKeyboardsList) {
```

Comment:
This should exclude (or gray out) languages that are already installed. Ideally, it'd show a greyed out entry with a check mark next to the language, to tell the user that they've already added the language.

Reply:
Right now, the list that comes from [kmpProcessor.getLanguageList()](https://github.com/keymanapp/keyman/blob/8e5d29557f580c6837c0432ef0db41b53a29f8f5/android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/PackageProcessor.java#L443-L447) already excludes installed languages.

Would you prefer I return the entire language list, and gray out installed ones?

Reply:
I see. That's a little unexpected on the part of kmpProcessor.getLanguageList() from my perspective -- it might be good to have a parameter to clarify that that happens?

What happens if the user has already installed all languages for a given keyboard? They need some feedback then.

Reply:
> What happens if the user has already installed all languages for a given keyboard? They need some feedback then.

Oh right, I meant to have the menu display a disabled entry that says something like "All languages installed"

Reply:
New changes:
* added boolean `excludeInstalledLanguages` that controls whether the the language list returns the entire list, or just the languages not installed.
* Installed languages grayed out with a check (disabled)

Reply:
* Title is different if all languages already installed
(see updated screenshot in PR description)